R/drive_read.R

```diff
@@ -0,0 +1,298 @@
+#' Read a Drive file
+#'
+#' @description This function reads the content of a Drive file into memory.
+#'   Some basic processing is available, but you might need to do additional
+#'   work to turn the content into an R object that is useful to you.
+#'
+#'   [drive_download()] is the more generally useful function, but for certain
+#'   file types, such as comma-separated values (MIME type `text/csv`), it can
+#'   be handy to read data directly from Google Drive and avoid writing to disk.
+#'
+#'   Just as for [drive_download()], native Google file types, such as Google
+#'   Sheets or Docs, must be exported as a conventional MIME type. See the help
+#'   for [drive_download()] for more about how that works and the examples
+#'   below.
+
+#' @template file-singular
+#' @inheritParams drive_download
+#' @param as How to process the response body.
+#'   * "auto": Determine from the MIME type of the response.
+#'     - "string" for `text/csv`, `text/plain`, or `text/tab-separated-values`
+#'     - "json" for `application/json`
+#'     - "html" for `text/html`
+#'     - "xml" for `application/xml` or `text/xml`
+#'     - "raw" otherwise
+#'   * "string": `httr::content(x, as = "text", ...)`
+#'   * "json": `httr::content(x, as = "parsed", type = "application/json", ...)`
+#'     which calls `jsonlite::fromJSON()`
+#'   * "html": `httr::content(x, as = "parsed", type = "text/html", ...)` which
+#'     calls `xml2::read_html()`
+#'   * "xml": `httr::content(x, as = "parsed", type = "text/xml", ...)` which
+#'     calls `xml2::read_xml()`
+#'   * "raw": `httr::content(x, as = "raw")`
+
+#' @param ... Additional parameters passed along to body processing functions,
+#'   for example, the `encoding` of the input `file`.
+#'
+
+#' @return The content of `file`, in the form specified via `as`. When `as =
+#'   "string"`, the content is (re-)encoded as UTF-8, which is also true, by
+#'   definition, for `as = "json"`. Any `encoding` parameter provided via `...`
+#'   is understood to describe the *input* `file`.
```

Comment:
As usual, it's kind of hard to explain what will happen encoding-wise, without really getting into the weeds.